power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -36,16 +39,125 @@ static_assert(is_indptr_mutable_v<writable_dataset_t>);
 struct ComponentInfo {
     MetaComponent const* component;
     // for non-uniform component, this is -1, we use indptr to describe the elements per scenario
-    Idx elements_per_scenario;
-    Idx total_elements;
+    Idx elements_per_scenario{};
+    Idx total_elements{};
 };
 
 struct DatasetInfo {
-    bool is_batch;
-    Idx batch_size; // for single dataset, the batch size is one
-    MetaDataset const* dataset;
+    bool is_batch{false};
+    Idx batch_size{0}; // for single dataset, the batch size is one
```

Comment:
I think the default `batch_size` should be one because you set `is_batch` default to `false`.

Reply:
Correct. Done.